File: src/main/java/data/model/MyRandom.java
package data.model;

class MyRandom {
    private long prevValue;

    MyRandom(int seed) {
        prevValue = seed* 25923;
    }
    double nextDouble(){
        long value = prevValue * 6364136223846723L;
        long shift = ((value >>> 16L) ^ prevValue) >>> 27L;
        long rot = prevValue >>> 59L;
        value =  (shift >>> rot) | (shift << ((-rot) & 31L));
        prevValue = value;
        return value;
    }
}

File: src/main/java/data/model/MyRandom.kt
package data.model

class MyRandom(seed: Int) {
    private var prevValue = seed * 25923L

    fun nextDouble(): Double {
        var value = prevValue * 6364136223846723L
        val shift = ((value ushr 16) xor prevValue) ushr 27
        val rot = prevValue ushr 59
        value = (shift ushr rot.toInt()) or (shift shl (-rot.toInt() and 31))
        prevValue = value
        return value.toDouble()
    }
}
